MovieCave - MVVM/MovieCave/MovieCave/Custom UI/CollectionViewReusableCell/CollectionViewReusableCell.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  CollectionViewCell.swift
+//  MovieCave
+//
+//  Created by Admin on 28.09.23.
+//
+
+import UIKit
+
+protocol CollectionViewReusableCellProtocol {
+    /// Configures the cell to display data for a TV show.
+    /// - Parameter series: The TV show data to display.
+    func setTVSeries(with series: TVSeriesResults)
+    
+    /// Configures the cell to display data for a movie.
+    /// - Parameter movies: The movie data to display.
+    func setMovies(with movies: MoviesModel)
+}
+
+protocol FavoriteButtonDelegate: AnyObject {
+    
+    /// Tells the delegate that the favorite button was clicked for the item at the given index.
+    /// - Parameter index: The index of the item whose favorite button was clicked.
+    func buttonClicked(index: Int)
+}
+
+class CollectionViewReusableCell: UICollectionViewCell, CollectionViewReusableCellProtocol {
+    
+    //MARK: - Properties
+    weak var delegate: FavoriteButtonDelegate?
+    private var favoriteButton: UIButton!
+    private let emptyStar = UIImage(systemName: Constants.starImage)?
+        .resizeSystemImage(sizeChange: CGSize(width: Constants.systemIconWidht, height: Constants.systemIconHeight), tintColor: .systemYellow)
+    private let filledStar = UIImage(systemName: Constants.starFillImage)?
+        .resizeSystemImage(sizeChange: CGSize(width: Constants.systemIconWidht, height: Constants.systemIconHeight), tintColor: .systemYellow)
+    private var imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = .scaleAspectFill
+        imageView.clipsToBounds = true
+        return imageView
+    }()
+    var movieIindex: Int?
+    
+    
+    func setTVSeries(with series: TVSeriesResults) {
```

Comment:
Is it possible to add two different protocols here. One for the setTVSeries and one for the setMovies?